### PR TITLE
Fix convert root to basenode

### PR DESCRIPTION
**Pull request by BenjaminDeleze**
Add the SubProto of a Proto to the extern Proto list when clicking on `Convert root to base node`

PS: I am not sure that the ways I modified `combinePaths` is very clean.

### CODE REVIEW

**Review by ad-daniel — CHANGES_REQUESTED** [review]
The implementation looks good, I've only had a warning on this:
- `highway_overtake.wbt`: convert `StraightRoadSegment.proto`

File: src/webots/vrml/WbNode.cpp [hunk 1]
```diff
@@ -1214,6 +1219,55 @@ void WbNode::exportNodeContents(WbWriter &writer) const {
...
+    if (match.hasMatch()) {
+      const QString subProto = match.captured(1);
+      const QString url = path;
+      QString error;
+      const QString subProtoUrl = WbProtoTreeItem::combinePaths(subProto, path, &error);
+      if (!error.isEmpty()) {
+        parsingWarn(error);
+        continue;
+      }
+      if (!subProtoUrl.endsWith(".proto")) {
```
Comment (ad-daniel) [inline src/webots/vrml/WbNode.cpp:1252]:
```suggestion
      if (!subProtoUrl.endsWith(".proto", Qt::CaseInsensitive)) {
```

File: src/webots/vrml/WbProtoTreeItem.cpp [hunk 2]
```diff
@@ -63,6 +63,7 @@ void WbProtoTreeItem::parseItem() {
     QRegularExpressionMatch match = it.next();
     if (match.hasMatch()) {
       const QString subProto = match.captured(1);
+
```
Comment (ad-daniel) [inline src/webots/vrml/WbProtoTreeItem.cpp:66]:
```suggestion
```

File: src/webots/vrml/WbNode.cpp [hunk 3]
```diff
@@ -26,12 +26,15 @@
...
 #include "WbMFVector3.hpp"
+#include "WbNetwork.hpp"
 #include "WbNodeFactory.hpp"
 #include "WbNodeModel.hpp"
 #include "WbNodeReader.hpp"
 #include "WbParser.hpp"
 #include "WbProject.hpp"
+#include "WbProtoManager.hpp"
 #include "WbProtoModel.hpp"
+#include "WbProtoTreeItem.hpp"
```
Comment (ad-daniel) [inline src/webots/vrml/WbNode.cpp:37]:
```suggestion
```

Comment (ad-daniel) [reply src/webots/vrml/WbNode.cpp:37]: Since we moved `combinePath` to WbUrl, I don't think this import is necessary anymore

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Just details, otherwise it seems to work very well

**Comment by BenjaminDeleze** [comment]
Everything seems ok for me

**Pull request merged**